sdks/java/extensions/sql/jdbc/src/main/java/org/apache/beam/sdk/extensions/sql/jdbc/BeamSqlLine.java

```diff
@@ -19,41 +19,57 @@
 
 import static org.apache.beam.sdk.extensions.sql.impl.JdbcDriver.CONNECT_STRING_PREFIX;
 
-import com.google.common.collect.ImmutableList;
 import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import javax.annotation.Nullable;
 import sqlline.SqlLine;
+import sqlline.SqlLine.Status;
 
 /** {@link BeamSqlLine} provides default arguments to SqlLine. */
 public class BeamSqlLine {
 
   private static final String NICKNAME = "BeamSQL";
 
   public static void main(String[] args) throws IOException {
+    runSqlLine(checkConnectionArgs(args), null, null, null);
+  }
+
+  private static String[] checkConnectionArgs(String[] args) {
+    List<String> argsList = new ArrayList<String>(Arrays.asList(args));
 
-    // Until we learn otherwise, we expect to add -nn <nickname> -u <url>
-    @Nullable String databaseUrl = null;
-    @Nullable String nickname = null;
-
-    // Provide -u and -nn only if they do not exist
-    for (int i = 0; i < args.length; i++) {
-      if (args[i].equals("-u")) {
-        databaseUrl = args[++i];
-      } else if (args[i].equals("-nn")) {
-        nickname = args[++i];
-      }
+    if (!argsList.contains("-nn")) {
+      argsList.add("-nn");
+      argsList.add(NICKNAME);
     }
 
-    ImmutableList.Builder wrappedArgs = ImmutableList.builder().addAll(Arrays.asList(args));
-    if (databaseUrl == null) {
-      wrappedArgs.add("-u").add(CONNECT_STRING_PREFIX);
+    if (!argsList.contains("-u")) {
+      argsList.add("-u");
+      argsList.add(CONNECT_STRING_PREFIX);
     }
-    if (nickname == null) {
-      wrappedArgs.add("-nn").add(NICKNAME);
+
+    return argsList.toArray(new String[argsList.size()]);
+  }
+
+  static Status runSqlLine(
+      String[] args,
+      InputStream inputStream,
+      @Nullable PrintStream outputStream,
+      @Nullable PrintStream errorStream)
```

Comment:
i think it makes sense to accept `OutputStream` instead of `PrintStream`, every consumer of this method will need to wrap it. I think it's better to have one place in `main` to specify `System.out` etc than have to create `PrintStream` in every test.

Reply:
Use `OutputStream` now. This is a good idea to simplify tests.